partner_location_auto_create/models/res_partner.py

```diff
@@ -108,14 +94,15 @@ def _remove_locations(self):
         by mistake.
         """
         move_obj = self.env['stock.move']
-        for location in self.location_ids:
-            moves = move_obj.search([
-                '|',
-                ('location_id', 'child_of', location.id),
-                ('location_dest_id', 'child_of', location.id),
-            ])
-            if not moves:
-                location.unlink()
+        for partner in self:
+            for location in partner.location_ids:
+                moves = move_obj.search([
```

Comment:
If you have locations with huge moves, this could be a bottleneck.
I would have used search_count() instead or search with a limit of 1 before the loop and create a dict with result grouping by location id and then iterating.